pecos/core/xmc/linear_solver.hpp

```diff
@@ -161,7 +161,7 @@ struct SVMWorker {
 
             const auto& xi = X.get_row(i);
             QD[i] += do_dot_product(xi, xi) + (param.bias > 0 ? param.bias * param.bias : 0);
-            double coef = inst_info[i].y * alpha[i];
+            double coef = (double) inst_info[i].y * alpha[i];
```

Comment:
Would this cast affect overall run-time cost?

Reply:
I did a quick check on eurlex-4k and the difference of old/new model training time is not statistically significant.

Reply:
This cast is inside a for-loop, and `inst_info[i].y` and `alpha[i]`are of the same type as matrix dtype which is usually `float32`, which means there will be 2 x loop-length cast. 
My concern is that if this function is a bottle-neck in some cases, it would increase time cost and is not necessary, given that we almost never encounter overflow problem before.

Reply:
I think it should be fine, because the for loop will be just executed once, and its not related to the max_iter in the optimization solver, which is really the main bottleneck of SVM solver.

Reply:
Got it. Thanks for the reply.